server/src/main/java/org/opensearch/index/remote/transfer/DownloadManager.java

```diff
@@ -6,47 +6,105 @@
-import org.opensearch.common.logging.Loggers;
+import org.opensearch.common.blobstore.BlobContainer;
+import org.opensearch.common.lucene.store.ByteArrayIndexInput;
 import org.opensearch.common.util.CancellableThreads;
 import org.opensearch.core.action.ActionListener;
-import org.opensearch.core.index.shard.ShardId;
+import org.opensearch.index.store.RemoteSegmentStoreDirectory;
+import org.opensearch.index.translog.transfer.TranslogTransferManager;
+import org.opensearch.index.translog.transfer.TranslogTransferMetadata;
 import org.opensearch.indices.recovery.RecoverySettings;
 import org.opensearch.threadpool.ThreadPool;
 
 import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
 import java.util.Collection;
 import java.util.Queue;
 import java.util.concurrent.ConcurrentLinkedQueue;
 import java.util.concurrent.ExecutionException;
 
 /**
- * Helper class to downloads files from a {@link RemoteSegmentStoreDirectory}
- * instance to a local {@link Directory} instance in parallel depending on thread
- * pool size and recovery settings.
+ * Manages file downloads from the remote store and from a {@link RemoteSegmentStoreDirectory}
+ * instance to a local {@link Directory} instance.
  *
  * @opensearch.api
  */
-@PublicApi(since = "2.11.0")
-public final class RemoteStoreFileDownloader {
-    private final Logger logger;
+@PublicApi(since = "2.13.0")
+public class DownloadManager {
+
     private final ThreadPool threadPool;
     private final RecoverySettings recoverySettings;
+    private static final Logger logger = LogManager.getLogger(DownloadManager.class);
 
-    public RemoteStoreFileDownloader(ShardId shardId, ThreadPool threadPool, RecoverySettings recoverySettings) {
-        this.logger = Loggers.getLogger(RemoteStoreFileDownloader.class, shardId);
+    public DownloadManager(ThreadPool threadPool, RecoverySettings recoverySettings) {
         this.threadPool = threadPool;
         this.recoverySettings = recoverySettings;
     }
 
+    public DownloadManager(DownloadManager other) {
+        this.threadPool = other.threadPool;
+        this.recoverySettings = other.recoverySettings;
+    }
+
+    /**
+     * Downloads the {@link TranslogTransferMetadata} file from the blob container stored as a part of remote store
+     * metadata
+     * @param blobContainer location where the metadata file is stored
+     * @param fileName name of the metadata file
+     * @return deserialized metadata object
+     * @throws IOException exception when reading from the remote file
+     */
+    public TranslogTransferMetadata fetchTranslogTransferMetadata(BlobContainer blobContainer, String fileName) throws IOException {
+        byte[] data = downloadFileFromRemoteStoreToMemory(blobContainer, fileName);
+        IndexInput indexInput = new ByteArrayIndexInput("metadata file", data);
+        return TranslogTransferManager.METADATA_STREAM_WRAPPER.readStream(indexInput);
```

Comment:
This creates a circular dependency with the TranslogTransferManager.  Can the translog manager invoke `downloadFileFromRemoteStoreToMemory` directly?

Reply:
This was removed.